### PR TITLE
Init Codecov

### PR DESCRIPTION
**Pull request by tigercosmos**
1. Please log in [codecov](https://about.codecov.io/), select the `modmesh` repo, and get the secret token.
2. enter the secret token in this repo setting
3. this PR has already been tested in the forked repo ([here](https://github.com/tigercosmos/modmesh/pull/5))
4. once this PR merge, later PR codecov will start to work
5. This PR focus on python coverage first

TODOs:
- C++ coverage
- C++ python binding coverage

### CODE REVIEW

**Comment by tigercosmos** [comment]
@yungyuc  please review.

**Comment by yungyuc** [comment]
> @yungyuc  please review.

Please make clear comments in the PR and let me know when it is done.

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,7 @@ DEBUG_SYMBOL ?= ON
 MODMESH_PROFILE ?= OFF
 BUILD_METAL ?= OFF
 BUILD_QT ?= ON
+COVERAGE ?= OFF
```
Comment (tigercosmos) [inline Makefile:27]: the flag for generate coverage report

File: Makefile [hunk 2]
```diff
@@ -70,6 +71,10 @@ else
 	PYTEST_OPTS ?=
 endif
 
+ifeq ($(COVERAGE), ON)
+	PYTEST_OPTS += --cov --cov-report=xml
```
Comment (tigercosmos) [inline Makefile:75]: generate report for pytest

File: .github/workflows/codecov.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+              run: |
+                rm -f build/*/Makefile
+                make cmake \
+                  VERBOSE=1 USE_CLANG_TIDY=OFF \
+                  BUILD_QT=OFF \
+                  CMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }} \
+                  CMAKE_ARGS="-DPYTHON_EXECUTABLE=$(which python3)"
+                make buildext VERBOSE=1
+        
+            - name: make pytest COVERAGE=ON BUILD_QT=OFF
```
Comment (tigercosmos) [inline .github/workflows/codecov.yml:89]: mainly copy from other yml file. note we have  `COVERAGE=ON` here

File: .github/workflows/codecov.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              run: |
+                python3 -c "import modmesh; assert modmesh.HAS_VIEW == False"
+                make pytest COVERAGE=ON VERBOSE=1
```
Comment (tigercosmos) [inline .github/workflows/codecov.yml:92]: run pytest and get  `coverage.xml` file

File: .github/workflows/codecov.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                
+            - name: Upload coverage reports to Codecov with GitHub Action
+              uses: codecov/codecov-action@v3
```
Comment (tigercosmos) [inline .github/workflows/codecov.yml:95]: codecov will read the `coverage.xml` file, and show on result on PR.

**Comment by tigercosmos** [comment]
@yungyuc  I believe now it is clearer. Please take a look

**Comment by yungyuc** [comment]
> @yungyuc I believe now it is clearer. Please take a look

After your annotations, I still get zero understanding about the tool and why do you want to add it.  I am afraid you may need to start from the 101.

**Comment by tigercosmos** [comment]
The PR is still unclear, close for now

**Pull request closed without merging**